infrastructures/docker/src/main/java/org/eclipse/che/workspace/infrastructure/docker/DockerInternalRuntime.java

```diff
@@ -378,8 +474,11 @@ public StartSynchronizer() {
             this.machines = new HashMap<>();
         }
 
-        public synchronized Map<String, ? extends Machine> getMachines() {
-            return Collections.unmodifiableMap(machines);
+        public synchronized Map<String, ? extends DockerMachine> getMachines() {
+            if (machines != null) {
+                return Collections.unmodifiableMap(machines);
```

Comment:
This method is used only in DockerInternalRuntime, so I think we can skip wrapping machines to unmodifiableMap. BTW I'd inline this with ternary operator :smile: